src/identity-providers/routes/AddMapper.ts

```diff
@@ -0,0 +1,26 @@
+import type { LocationDescriptorObject } from "history";
+import { generatePath } from "react-router-dom";
+import type { RouteDef } from "../../route-config";
+import { AddMapper } from "../add/AddMapper";
+
+export type IdentityProviderTab = "mappers";
+
+export type AddIdentityProviderMapperTabParams = {
+  realm: string;
+  id?: string;
+  tab?: IdentityProviderTab;
+  alias?: string;
+  providerId?: string;
```

Comment:
The `providerId` parameter is required in the path of the route so it should be here too.
```suggestion
  providerId: string;
```